src/components/combobox-item/combobox-item.tsx

```diff
@@ -203,8 +216,9 @@ export class ComboboxItem implements ConditionalSlotComponent, InteractiveCompon
           style={{ "--calcite-combobox-item-spacing-indent-multiplier": `${depth}` }}
         >
           <li class={classes} id={this.guid} onClick={this.itemClickHandler}>
-            {this.renderIcon(isSingleSelect)}
-            <span class={CSS.title}>{this.textLabel}</span>
+            {this.renderSelectIndicator(showDot, iconPath)}
```

Comment:
Both the indicator and icon are dynamically rendered, so [per Stencil doc](https://stenciljs.com/docs/templating-jsx#:~:text=If%20this%20array%20of%20children%20is%20dynamic%2C%20i.e.%2C%20if%20any%20nodes%20may%20be%20added%2C%20removed%2C%20or%20reordered%2C%20then%20it%27s%20a%20good%20idea%20to%20set%20a%20unique%20key%20attribute%20on%20each%20element%20like%20so%3A), can you make sure to add `key` to the root icon and indicator nodes?